mPower2/mPower2/View Controllers/TaskBrowserViewController.swift

```diff
@@ -0,0 +1,366 @@
+        vc.delegate = self
+        self.present(vc, animated: true, completion: nil)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        
+        guard let selectedTaskGroup = selectedTaskGroup else {
+            return UIEdgeInsetsMake(0, 0, 0, 0)
+        }
+        
+        // We do this so the cells are centered horizontally
+
+        let totalCellWidth = cellSize.width * CGFloat(selectedTaskGroup.tasks.count)
+        let totalSpacingWidth = minCellSpacing * (CGFloat(selectedTaskGroup.tasks.count) - 1)
+        let inset = (collectionView.frame.size.width - CGFloat(totalCellWidth + totalSpacingWidth)) / 2
+
+        return UIEdgeInsetsMake(kTopInset, inset, 0, inset)
+    }
+}
+
+protocol TaskBrowserTabViewDelegate {
+    func taskGroupSelected(identifier: String)
+}
+
+class TaskBrowserTabView: UIView {
+    
+    public var taskGroupIdentifier: String?
+    public var delegate: TaskBrowserTabViewDelegate?
+    
+    public var title: String? {
+        didSet {
+            label.text = title
+        }
+    }
+    public var isSelected: Bool = false {
+        didSet {
+            rule.isHidden = !isSelected
+        }
+    }
+    
+    private let rule: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = UIColor.royal500
+        return view
+    }()
+    private let label: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.boldSystemFont(ofSize: 16.0)
+        label.numberOfLines = 0
+        label.textColor = UIColor.darkGray
+        label.textAlignment = .center
+        return label
+    }()
+
+    public init(frame: CGRect, taskGroupIdentifier: String) {
+        super.init(frame: frame)
+        
+        self.taskGroupIdentifier = taskGroupIdentifier
+        
+        // Add our label
+        addSubview(label)
+        label.rsd_alignAllToSuperview(padding: 0.0)
+        
+        // Add our rule
+        addSubview(rule)
+        rule.rsd_makeHeight(.equal, 4.0)
+        rule.rsd_alignToSuperview([.leading, .trailing, .bottom], padding: 0.0)
+        
+        // Add a button
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addTarget(self, action: #selector(tabSelected), for: .touchUpInside)
+        addSubview(button)
+        button.rsd_alignAllToSuperview(padding: 0.0)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    @objc func tabSelected() {
+        if let delegate = delegate,
+            let taskGroupIdentifier = taskGroupIdentifier {
+            delegate.taskGroupSelected(identifier: taskGroupIdentifier)
+        }
+    }
+}
+
+class TaskCollectionViewCell: UICollectionViewCell {
```

Comment:
@IBDesignable and use the storyboard, please.

Reply:
done